### DISCUSSION TITLE
Will I need to re-pair my devices after upgrading the controller firmware?

### DISCUSSION THREAD

**Discussion by bushvin** [zwave-js/zwave-js-ui#2808] [Q&A] [answered]
Hi,

I've been trying to postpone [upgrading my controller firmware](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/otw-upgrade), but at some point you have to go with the flow.

If/When I perform these steps, will I be expected to re-pair all my devices? It's not entirely clear to me.

Thanks in advance

**Comment by jmgiaever** [comment, marked as answer]
No, not if everything goes well. But a backup of the device is advised.
